### PR TITLE
[idea] improved completion context matching

### PR DESCRIPTION
**Pull request by svenefftinge**
The pattern-based approach had way too many false positive matches, so I replaced 
them by our parser-based follow element computation. Instead of passing a pattern, you now directly pass abstract elements or eStructuralFeatures.

Advantages over the ContentAssistContextFactory-hook, is that the PsiFile’s XtextResource is not resolved and that we have the full IntelliJ API available.

### CODE REVIEW

File: intellij/org.eclipse.xtend.idea/src/org/eclipse/xtend/core/idea/completion/XtendCompletionContributor.xtend [hunk 1]
```diff
@@ -58,7 +56,7 @@ class XtendCompletionContributor extends AbstractXtendCompletionContributor {
 	protected def completeAbstractSuperMethods() {
 		extend(
 			CompletionType.BASIC,
-			psiElement.withEReference(XtendPackage.Literals.XTEND_TYPE_DECLARATION__MEMBERS)
+			grammarAccess.memberAccess.newKeyword_2_2_2
```
Comment (szarnekow) [inline intellij/org.eclipse.xtend.idea/src/org/eclipse/xtend/core/idea/completion/XtendCompletionContributor.xtend]: A little hacky - would it work to pass the EStructuralFeature XtendType.members here, instead?

Comment (svenefftinge) [inline intellij/org.eclipse.xtend.idea/src/org/eclipse/xtend/core/idea/completion/XtendCompletionContributor.xtend]: I changed it to use the feature. As it is just computing the elements once on startup, I don't think it is a problem performance-wise. 
Changing it revealed a serious bug, which is fixed in the updated commit. thx.

**Comment by svenefftinge** [comment]
Jenkins intellij-tests-master-latest

**Comment by svenefftinge** [comment]
Jenkins intellij-tests-master-latest

**Pull request merged**